edge-util/src/Microsoft.Azure.Devices.Edge.Storage/InMemoryDbStoreProvider.cs

```diff
@@ -9,11 +9,28 @@ public class InMemoryDbStoreProvider : IDbStoreProvider
     {
         const string DefaultPartitionName = "$Default";
         readonly ConcurrentDictionary<string, IDbStore> partitionDbStoreDictionary = new ConcurrentDictionary<string, IDbStore>();
+        readonly Option<IStorageSpaceChecker> memoryStorageSpaceChecker;
+
+        public InMemoryDbStoreProvider()
+            : this(Option.None<IStorageSpaceChecker>())
+        {
```

Comment:
What's the case when we want to set the checker to None? I see if the limit is not set this means is disabled, isn't that enough?

Reply:
Yeah, I just wanted to keep the parameterless constructor for the sake of Edge Agent for now. We can remove it later. Edge Agent currently doesn't use any of the checker stuff and just uses the default parameterless constructor

Reply:
Yeah, I kept it only for the sake of Edge Agent where we don't need to create and register an IStorageSpaceChecker that's passed to the InMemoryDbStoreProvider. We can later remove the Option and just use IStorageSpaceChecker directly